textattack/transformations/transformation.py

```diff
@@ -25,6 +25,7 @@ def __call__(self, current_text, pre_transformation_constraints=[], indices_to_m
             indices_to_modify = set(indices_to_modify)
         for constraint in pre_transformation_constraints:
             indices_to_modify = indices_to_modify & constraint(current_text, self)
+        indices_to_modify = current_text.convert_from_original_idxs(indices_to_modify)
```

Comment:
I think this method should assume `indices_to_modify` are up-to-date indices unless indicated otherwise, and have a boolean flag to indicate they're not. And this conversion definitely be above the for loop because the constraints act on the current indices not the original ones.

Reply:
yeah, the boolean is at the beginning of `tt.convert_from_original_idxs`. So it doesn't do anything expensive unless `len(tt.attack_attrs['original_modified_indices']) > 0`. Good call about moving the conversion up

Reply:
What boolean?

Reply:
https://github.com/QData/TextAttack/blob/ec857cb59f3233479747ecc9519eae363920bf28/textattack/shared/tokenized_text.py#L159

Reply:
You mean when there's no `original_index_map`? I'm trying to say that the default behavior should be to assume that the argument `indices_to_modify` represents the current rather than original indices. In our case our current things pass in original indices but I don't think that makes sense as the default. So it has nothing to do with whether the TT has an `original_index_map` yet.

Reply:
In the default case (no shifts) `original_index_map` won't exist so convert_from_original_idxs will return the same object that was passed in. I don't see anywhere where the original indices are passed in.